2016-05/may-19.md

```diff
@@ -0,0 +1,65 @@
+## May 19 ([discuss](https://github.com/reactjs/core-notes/pull/15))
+
+### Attendees
+
+* [Ben](https://twitter.com/soprano) (React)
+* [Christopher](https://twitter.com/vjeux) (React Native)
+* [Dan](https://twitter.com/dan_abramov) (React)
+* [Jim](http://github.com/jimfb) (React)
+* [Keyan](https://twitter.com/keyanzhang) (React, intern)
+* [Paul](https://twitter.com/zpao) (React)
+* [Sebastian](https://twitter.com/sebmarkbage) (React)
+* [Tom](https://twitter.com/tomocchino) (React)
+
+### `createClass()` vs ES2015 Classes
+
+#### Why Use Classes?
+
+* `createClass()` API feels outdated, and can be implemented entirely in userland if needed.
+* ES6 classes are less of an API, possibly simpler.
+* ES6 classes are friendlier to static analysis.
+* ES6 classes offer performance improvements in large apps.
+* We are in the business of UI components, not type systems.
+
+#### What’s Holding Us Back?
+
+##### Autobinding
+
+* [Class properties](https://github.com/jeffmo/es-class-fields-and-static-properties) solve this.
+* However they’re not moving forward at TC39 yet as [Jeff](https://github.com/jeffmo) is busy preparing for React Europe.
+* Downside: nobody wants experimental syntax in the documentation.
+* We could just always bind in constructor in the documentation.
+
+##### Mixins
+
+* A lot of components on Facebook website still use mixins, but mostly the same ones.
+* If we decided to deprecate `createClass` in the future, we would need to fix them first.
+* Possible temporary workaround: apply them on top of ES6 class like [react-mixin](https://github.com/brigand/react-mixin) does.
+
+#### Next Steps
+
+* We should update the documentation to use ES6 classes and functional components wherever possible.
+* Eventually we should deprecate `createClass`, possibly move it into a separate package.
+* We need to have a good solution for folks who don’t want to use a transpiler.
+
+### Separating Out `PropTypes`
+
+* [Jordan](https://twitter.com/ljharb) and some Airbnb folks asked about separating `PropTypes` into a generic validation library.
+* One concern might be that as a separate project `PropTypes` could become incompatible (as a type system) with Flow or TypeScript.
```

Comment:
Is PropTypes validation removed when compiling with NODE_ENV set to production?
If not, then while visiting this topic it would be an interesting constraint to add.